src/dotnet/BuiltInCommandsCatalog.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.DotNet.Tools.Add;
+using Microsoft.DotNet.Tools.Build;
+using Microsoft.DotNet.Tools.Clean;
+using Microsoft.DotNet.Tools.Help;
+using Microsoft.DotNet.Tools.List;
+using Microsoft.DotNet.Tools.Migrate;
+using Microsoft.DotNet.Tools.MSBuild;
+using Microsoft.DotNet.Tools.New;
+using Microsoft.DotNet.Tools.NuGet;
+using Microsoft.DotNet.Tools.Pack;
+using Microsoft.DotNet.Tools.Publish;
+using Microsoft.DotNet.Tools.Remove;
+using Microsoft.DotNet.Tools.Restore;
+using Microsoft.DotNet.Tools.Run;
+using Microsoft.DotNet.Tools.Sln;
+using Microsoft.DotNet.Tools.Test;
+using Microsoft.DotNet.Tools.VSTest;
+using Microsoft.DotNet.Tools.Cache;
+
+namespace Microsoft.DotNet.Cli
+{
+    public static class BuiltInCommandsCatalog
+    {
+        public static Dictionary<string, BuiltInCommandMetadata> Commands = new Dictionary<string, BuiltInCommandMetadata>
+        {
+            ["add"] = new BuiltInCommandMetadata
```

Comment:
@jonsequitur this sounds like a good idea, but see below on the need to possibly keep this information in a separate file. Not saying that this would be impossible in the new parser, of course. :)